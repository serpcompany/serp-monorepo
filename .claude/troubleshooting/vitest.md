https://vitest.dev/guide/common-errors.html