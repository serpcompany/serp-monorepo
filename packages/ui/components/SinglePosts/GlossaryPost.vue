<template>
  <div>
    <div class="mt-10 grid grid-cols-1 lg:grid-cols-3">
      <div class="col-span-2 pb-10">
        <h1 class="text-4xl font-bold lg:text-6xl">{{ data.title }}</h1>
        <p v-if="data.oneLiner" class="prose dark:prose-invert mt-4">
          {{ data.oneLiner }}
        </p>
      </div>
      <div class="col-span-1 pb-10">
        <s-script-you-tube-player
          v-if="data.videoId"
          ref="video"
          :video-id="data.videoId"
          @ready="isLoaded = true"
          @state-change="stateChange"
        >
          <template #awaitingLoad>
            <div
              class="absolute top-1/2 left-1/2 h-[48px] w-[68px] -translate-x-1/2 -translate-y-1/2 transform"
            >
              <svg height="100%" version="1.1" viewBox="0 0 68 48" width="100%">
                <path
                  d="M66.52,7.74c-0.78-2.93-2.49-5.41-5.42-6.19C55.79,.13,34,0,34,0S12.21,.13,6.9,1.55 C3.97,2.33,2.27,4.81,1.48,7.74C0.06,13.05,0,24,0,24s0.06,10.95,1.48,16.26c0.78,2.93,2.49,5.41,5.42,6.19 C12.21,47.87,34,48,34,48s21.79-0.13,27.1-1.55c2.93-0.78,4.64-3.26,5.42-6.19C67.94,34.95,68,24,68,24S67.94,13.05,66.52,7.74z"
                  fill="#f00"
                />
                <path d="M 45,24 27,14 27,34" fill="#fff" />
              </svg>
            </div>
          </template>
        </s-script-you-tube-player>
      </div>

      <u-separator class="col-span-3 my-4">
        <s-logo />
      </u-separator>

      <!-- main section -->
      <div class="prose dark:prose-invert col-span-2">
        <article v-html="data.content"></article>
        <div v-if="data.relatedPosts && data.relatedPosts.length === 0">
          <h2>Related Posts</h2>
          <ul>
            <li>Term 1</li>
            <li>Term 2</li>
            <li>Term 3</li>
          </ul>
        </div>
      </div>

      <!-- sidebar -->
      <div class="col-span-1"></div>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { Post } from '@serp/types/types';

const isLoaded = ref(false);
const isPlaying = ref(false);
const video = ref();

defineProps<{
  data: Post;
}>();

function stateChange(event: { data: number }) {
  isPlaying.value = event.data === 1;
}
</script>
