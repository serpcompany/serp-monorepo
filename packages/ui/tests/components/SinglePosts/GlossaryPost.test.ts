/* eslint-disable @typescript-eslint/no-unused-vars, no-unused-vars  */

import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import GlossaryPost from '../../../components/SinglePosts/GlossaryPost.vue';
import ComponentRender from '../../componentRender';
import '../../mockUseUserSession';

let runtimeConfig: Record<string, unknown> = { public: { useAuth: true } };
mockNuxtImport('useHead', () => () => {});

describe('SinglePostsGlossaryPost Snapshot', () => {
  const scenarios: [
    string,
    { config: Record<string, unknown>; props: { data: unknown } }
  ][] = [
    [
      'with full post and auth enabled',
      {
        config: { app: { baseURL: '/' }, public: { useAuth: true } },
        props: {
          data: {
            // Using keyword when available (takes precedence over title)
            keyword: 'Nuxt Testing',
            oneLiner: 'A quick guide on snapshot testing.',
            videoId: 'abc123',
            content: '<p>Content of the post</p>',
            // When relatedPosts exists and is empty, a related posts block is rendered.
            relatedPosts: [],
            slug: 'nuxt-testing',
            comments: [
              { author: 'John Doe', comment: 'Great post!', replies: [] }
            ],
            upvotes: ['test@test.com']
          }
        }
      }
    ],
    [
      'with full post and auth disabled',
      {
        config: { app: { baseURL: '/' }, public: { useAuth: false } },
        props: {
          data: {
            keyword: 'Nuxt Testing',
            oneLiner: 'A quick guide on snapshot testing.',
            videoId: 'abc123',
            content: '<p>Content of the post</p>',
            relatedPosts: [],
            slug: 'nuxt-testing',
            comments: [
              { author: 'John Doe', comment: 'Great post!', replies: [] }
            ],
            upvotes: []
          }
        }
      }
    ],
    [
      'without video and minimal fields',
      {
        config: { app: { baseURL: '/' }, public: { useAuth: true } },
        props: {
          data: {
            // Omitting keyword, so title will be used for the heading.
            title: 'No Video Post',
            content: '<p>Content without video</p>',
            slug: 'no-video-post',
            // Comments and relatedPosts are omitted or empty.
            comments: [],
            upvotes: []
          }
        }
      }
    ],
    [
      'with non-empty relatedPosts',
      {
        config: { app: { baseURL: '/' }, public: { useAuth: true } },
        props: {
          data: {
            title: 'Related Posts Present',
            oneLiner: 'Post with related posts',
            videoId: 'def456',
            content: '<p>Some content here</p>',
            // Providing non-empty relatedPosts should bypass the related posts block.
            relatedPosts: [{ id: 1, title: 'Related Post' }],
            slug: 'related-posts-post',
            comments: [],
            upvotes: ['test@test.com']
          }
        }
      }
    ]
  ];

  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, { config, props }) => {
      runtimeConfig = config;
      mockNuxtImport('useRuntimeConfig', () => () => runtimeConfig);
      globalThis.usePostComments = async (id: number) => {
        return {
          comments: props.comments
            ? [
                {
                  id: 1,
                  content: 'Test comment',
                  replies: []
                }
              ]
            : []
        };
      };

      const html = await ComponentRender(
        `GlossaryPost ${desc}`,
        { props },
        GlossaryPost
      );
      expect(html).toMatchSnapshot();
    }
  );
});
