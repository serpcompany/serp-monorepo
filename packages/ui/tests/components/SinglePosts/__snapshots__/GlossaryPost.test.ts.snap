// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SinglePostsGlossaryPost Snapshot > renders with full post and auth disabled correctly 1`] = `
"<div>
  <div class="mt-10 grid grid-cols-1 lg:grid-cols-3">
    <div class="col-span-2 pb-10">
      <h1 class="text-4xl font-bold lg:text-6xl">Nuxt Testing</h1>
      <p class="prose dark:prose-invert mt-4">A quick guide on snapshot testing.</p>
    </div>
    <div class="col-span-1 pb-10">
      <div aria-busy="false" aria-label="YouTube Player - Placeholder" aria-live="polite" role="application" style="cursor: pointer; position: relative; background-color: black; max-width: 100%; width: 640px; height: auto; aspect-ratio: 640 / 360;">
        <div style="width: 100%; height: 100%; position: absolute; top: 0px; left: 0px;"></div><img src="https://i.ytimg.com/vi_webp/abc123/sddefault.webp" alt="" loading="lazy" style="width: 100%; object-fit: cover; height: 100%;">
        <!--v-if-->
        <div class="absolute top-1/2 left-1/2 h-[48px] w-[68px] -translate-x-1/2 -translate-y-1/2 transform"><svg height="100%" version="1.1" viewBox="0 0 68 48" width="100%">
            <path d="M66.52,7.74c-0.78-2.93-2.49-5.41-5.42-6.19C55.79,.13,34,0,34,0S12.21,.13,6.9,1.55 C3.97,2.33,2.27,4.81,1.48,7.74C0.06,13.05,0,24,0,24s0.06,10.95,1.48,16.26c0.78,2.93,2.49,5.41,5.42,6.19 C12.21,47.87,34,48,34,48s21.79-0.13,27.1-1.55c2.93-0.78,4.64-3.26,5.42-6.19C67.94,34.95,68,24,68,24S67.94,13.05,66.52,7.74z" fill="#f00"></path>
            <path d="M 45,24 27,14 27,34" fill="#fff"></path>
          </svg></div>
      </div>
    </div>
    <div data-orientation="horizontal" role="separator" class="flex items-center align-center text-center w-full flex-row col-span-3 my-4">
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
      <div class="font-medium text-(--ui-text) flex mx-3 whitespace-nowrap">
        <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
            <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
          </svg></div>
      </div>
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
    </div><!-- main section -->
    <div class="prose dark:prose-invert col-span-2">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <article>
        <p>Content of the post</p>
      </article>
      <div>
        <h2>Related Posts</h2>
        <ul>
          <li>Term 1</li>
          <li>Term 2</li>
          <li>Term 3</li>
        </ul>
      </div><!-- Comments Section -->
      <!--v-if-->
    </div><!-- sidebar -->
    <div class="col-span-1"></div>
  </div>
</div>"
`;

exports[`SinglePostsGlossaryPost Snapshot > renders with full post and auth enabled correctly 1`] = `
"<div>
  <div class="mt-10 grid grid-cols-1 lg:grid-cols-3">
    <div class="col-span-2 pb-10">
      <h1 class="text-4xl font-bold lg:text-6xl">Nuxt Testing</h1>
      <p class="prose dark:prose-invert mt-4">A quick guide on snapshot testing.</p>
    </div>
    <div class="col-span-1 pb-10">
      <div aria-busy="false" aria-label="YouTube Player - Placeholder" aria-live="polite" role="application" style="cursor: pointer; position: relative; background-color: black; max-width: 100%; width: 640px; height: auto; aspect-ratio: 640 / 360;">
        <div style="width: 100%; height: 100%; position: absolute; top: 0px; left: 0px;"></div><img src="https://i.ytimg.com/vi_webp/abc123/sddefault.webp" alt="" loading="lazy" style="width: 100%; object-fit: cover; height: 100%;">
        <!--v-if-->
        <div class="absolute top-1/2 left-1/2 h-[48px] w-[68px] -translate-x-1/2 -translate-y-1/2 transform"><svg height="100%" version="1.1" viewBox="0 0 68 48" width="100%">
            <path d="M66.52,7.74c-0.78-2.93-2.49-5.41-5.42-6.19C55.79,.13,34,0,34,0S12.21,.13,6.9,1.55 C3.97,2.33,2.27,4.81,1.48,7.74C0.06,13.05,0,24,0,24s0.06,10.95,1.48,16.26c0.78,2.93,2.49,5.41,5.42,6.19 C12.21,47.87,34,48,34,48s21.79-0.13,27.1-1.55c2.93-0.78,4.64-3.26,5.42-6.19C67.94,34.95,68,24,68,24S67.94,13.05,66.52,7.74z" fill="#f00"></path>
            <path d="M 45,24 27,14 27,34" fill="#fff"></path>
          </svg></div>
      </div>
    </div>
    <div data-orientation="horizontal" role="separator" class="flex items-center align-center text-center w-full flex-row col-span-3 my-4">
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
      <div class="font-medium text-(--ui-text) flex mx-3 whitespace-nowrap">
        <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
            <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
          </svg></div>
      </div>
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
    </div><!-- main section -->
    <div class="prose dark:prose-invert col-span-2">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <article>
        <p>Content of the post</p>
      </article>
      <div>
        <h2>Related Posts</h2>
        <ul>
          <li>Term 1</li>
          <li>Term 2</li>
          <li>Term 3</li>
        </ul>
      </div><!-- Comments Section -->
      <div class="mt-10">
        <h2 class="mb-6 text-3xl font-bold">Comments</h2>
        <div data-v-e7198dba="" class="comments comments-github-style" style="background: transparent;">
          <div data-v-e7198dba="" class="innerWrapper">
            <div data-v-e7198dba="" class="userName">
              <div data-v-e7198dba="" class="addComment">
                <div data-v-e7198dba="" class="w-12 rounded-t-full">
                  <!--v-if-->
                </div>
                <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
                  <!--v-if-->
                  <!--v-if-->
                </div>
              </div>
            </div>
            <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div><!-- sidebar -->
    <div class="col-span-1"></div>
  </div>
</div>"
`;

exports[`SinglePostsGlossaryPost Snapshot > renders with non-empty relatedPosts correctly 1`] = `
"<div>
  <div class="mt-10 grid grid-cols-1 lg:grid-cols-3">
    <div class="col-span-2 pb-10">
      <h1 class="text-4xl font-bold lg:text-6xl">Related Posts Present</h1>
      <p class="prose dark:prose-invert mt-4">Post with related posts</p>
    </div>
    <div class="col-span-1 pb-10">
      <div aria-busy="false" aria-label="YouTube Player - Placeholder" aria-live="polite" role="application" style="cursor: pointer; position: relative; background-color: black; max-width: 100%; width: 640px; height: auto; aspect-ratio: 640 / 360;">
        <div style="width: 100%; height: 100%; position: absolute; top: 0px; left: 0px;"></div><img src="https://i.ytimg.com/vi_webp/def456/sddefault.webp" alt="" loading="lazy" style="width: 100%; object-fit: cover; height: 100%;">
        <!--v-if-->
        <div class="absolute top-1/2 left-1/2 h-[48px] w-[68px] -translate-x-1/2 -translate-y-1/2 transform"><svg height="100%" version="1.1" viewBox="0 0 68 48" width="100%">
            <path d="M66.52,7.74c-0.78-2.93-2.49-5.41-5.42-6.19C55.79,.13,34,0,34,0S12.21,.13,6.9,1.55 C3.97,2.33,2.27,4.81,1.48,7.74C0.06,13.05,0,24,0,24s0.06,10.95,1.48,16.26c0.78,2.93,2.49,5.41,5.42,6.19 C12.21,47.87,34,48,34,48s21.79-0.13,27.1-1.55c2.93-0.78,4.64-3.26,5.42-6.19C67.94,34.95,68,24,68,24S67.94,13.05,66.52,7.74z" fill="#f00"></path>
            <path d="M 45,24 27,14 27,34" fill="#fff"></path>
          </svg></div>
      </div>
    </div>
    <div data-orientation="horizontal" role="separator" class="flex items-center align-center text-center w-full flex-row col-span-3 my-4">
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
      <div class="font-medium text-(--ui-text) flex mx-3 whitespace-nowrap">
        <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
            <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
          </svg></div>
      </div>
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
    </div><!-- main section -->
    <div class="prose dark:prose-invert col-span-2">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <article>
        <p>Some content here</p>
      </article>
      <!--v-if-->
      <!-- Comments Section -->
      <div class="mt-10">
        <h2 class="mb-6 text-3xl font-bold">Comments</h2>
        <div data-v-e7198dba="" class="comments comments-github-style" style="background: transparent;">
          <div data-v-e7198dba="" class="innerWrapper">
            <div data-v-e7198dba="" class="userName">
              <div data-v-e7198dba="" class="addComment">
                <div data-v-e7198dba="" class="w-12 rounded-t-full">
                  <!--v-if-->
                </div>
                <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
                  <!--v-if-->
                  <!--v-if-->
                </div>
              </div>
            </div>
            <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div><!-- sidebar -->
    <div class="col-span-1"></div>
  </div>
</div>"
`;

exports[`SinglePostsGlossaryPost Snapshot > renders without video and minimal fields correctly 1`] = `
"<div>
  <div class="mt-10 grid grid-cols-1 lg:grid-cols-3">
    <div class="col-span-2 pb-10">
      <h1 class="text-4xl font-bold lg:text-6xl">No Video Post</h1>
      <!--v-if-->
    </div>
    <div class="col-span-1 pb-10">
      <!--v-if-->
    </div>
    <div data-orientation="horizontal" role="separator" class="flex items-center align-center text-center w-full flex-row col-span-3 my-4">
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
      <div class="font-medium text-(--ui-text) flex mx-3 whitespace-nowrap">
        <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
            <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
          </svg></div>
      </div>
      <div class="border-(--ui-border) w-full border-solid border-t"></div>
    </div><!-- main section -->
    <div class="prose dark:prose-invert col-span-2">
      <!-- eslint-disable-next-line vue/no-v-html -->
      <article>
        <p>Content without video</p>
      </article>
      <!--v-if-->
      <!-- Comments Section -->
      <div class="mt-10">
        <h2 class="mb-6 text-3xl font-bold">Comments</h2>
        <div data-v-e7198dba="" class="comments comments-github-style" style="background: transparent;">
          <div data-v-e7198dba="" class="innerWrapper">
            <div data-v-e7198dba="" class="userName">
              <div data-v-e7198dba="" class="addComment">
                <div data-v-e7198dba="" class="w-12 rounded-t-full">
                  <!--v-if-->
                </div>
                <div data-v-e7198dba="" class="commentBox text-black dark:text-white"><textarea data-v-e7198dba="" name="addComment" class="addComment border-neutral-500 bg-white dark:border-neutral-400 dark:bg-neutral-800" placeholder="Add a comment" spellcheck="true" aria-label="Add comment"></textarea>
                  <!--v-if-->
                  <!--v-if-->
                </div>
              </div>
            </div>
            <transition-group-stub data-v-e7198dba="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div><!-- sidebar -->
    <div class="col-span-1"></div>
  </div>
</div>"
`;
